lib/json_schemer/resources.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module JSONSchemer
+  class Resources
+
+    def initialize
+      @resources ||= { :lexical => {}, :dynamic => {} }
+    end
+
+    def register(type, namespace_uri, schema)
+      @resources[type][namespace_uri.to_s] = schema
```

Comment:
```suggestion
      fail "Schema already registered as #{type} under #{namespace_uri}" if @resources[type][namespace_uri.to_s]

      @resources[type][namespace_uri.to_s] = schema
```
Check whether the URI is already used for possibly another schema. It makes a couple of tests fail though.